diff -BurN /home/tking/nook/distro/kernel/drivers/power/max17042_battery.c /home/tking/nook/distro/kernel-temp/drivers/power/max17042_battery.c
--- /home/tking/nook/distro/kernel/drivers/power/max17042_battery.c	2010-11-25 12:49:49.000000000 -0800
+++ /home/tking/nook/distro/kernel-temp/drivers/power/max17042_battery.c	2010-12-23 01:22:21.000000000 -0800
@@ -750,13 +750,13 @@
 		// in unit of micro-degree, 
 		if ( temp & (1<<15) ) {
 			//negtive value
-			max17042->temp_cached = COMPLEMENT_VAL(temp, TEMP_RESOLUTION, 1); 
-			//(~(temp & 0x7FFF) + 1) * TEMP_RESOLUTION * (-1);
+			max17042->temp_cached = COMPLEMENT_VAL(temp, TEMP_RESOLUTION, 100000); 
+			//(( ((((~temp) & 0x7FFF) + 1) * 3900)  / 100000 ) * (-1))
 		}
 		else {
 			//positive value
-			max17042->temp_cached = NON_COMPLEMENT_VAL(temp, TEMP_RESOLUTION, 1);
-			//(temp & 0x7FFF) * TEMP_RESOLUTION;
+			max17042->temp_cached = NON_COMPLEMENT_VAL(temp, TEMP_RESOLUTION, 100000);
+			//( ((((~temp) & 0x7FFF) + 1) * 3900)  / 100000 ) )
 		}
 	}
 	
@@ -1322,7 +1322,7 @@
 	max17042->rsoc_base = max17042->rsoc_cached = 100;
 	max17042->volt_cached = 4000000;
 	max17042->curr_cached = 300000;
-   	max17042->temp_cached = 25000000;	
+   	max17042->temp_cached = 250;	
 	max17042->dumpFormat = DUMP_FMT_NONE;
 	max17042->next_save = jiffies + HISTORY_REFRESH_INTERVAL*HZ;

